src/react/ISV/components/BucketField.tsx

```diff
@@ -0,0 +1,230 @@
+);
+
+type BucketFieldProps = {
+  bucketNameTooltip?: React.JSX.Element;
+  platform?: string;
+};
+
+type BucketField = {
+  name: string;
+  tag?: string;
+  capacity?: string;
+  capacityUnit?: string;
+};
+
+type FormValues = {
+  buckets: BucketField[];
+  bucketNumber?: number;
+};
+
+const BucketNameFormGroup = ({
+  index,
+  errors,
+  bucketNamePlaceholder,
+  bucketNumber,
+  bucketNameTooltip,
+  register,
+}: {
+  index: number;
+  errors: FieldErrors<FormValues>;
+  bucketNamePlaceholder: string;
+  bucketNumber: number;
+  bucketNameTooltip: React.JSX.Element;
+  register: UseFormRegister<FormValues>;
+}) => (
+  <FormGroup
+    id={`bucketName-${index}`}
+    label={bucketNumber > 1 ? `Bucket #${index + 1} name` : 'Bucket name'}
+    required
+    labelHelpTooltip={bucketNameTooltip}
+    error={(errors?.buckets?.[index]?.name?.message as string) ?? ''}
+    helpErrorPosition="bottom"
+    content={
+      <Input
+        id={`bucketName-${index}`}
+        type="text"
+        autoComplete="off"
+        placeholder={bucketNamePlaceholder}
+        {...register(`buckets.${index}.name`)}
+      />
+    }
+  />
+);
+
+const BucketField = (fieldOverrides: BucketFieldProps) => {
+  const { bucketNameTooltip = defaultBucketNameTooltip, platform } =
+    fieldOverrides;
+
+  const theme = useTheme();
+  const [bucketNumber, setBucketNumber] = useState<number>(1);
+
+  const {
+    register,
+    control,
+    formState: { errors },
+  } = useFormContext<FormValues>();
+
+  const { fields, append, remove, replace } = useFieldArray({
+    name: 'buckets',
+    control,
+  });
+
+  useEffect(() => {
+    if (fields.length === 0) {
+      append({
+        name: '',
+        tag: platform,
+        capacity: '0',
+        capacityUnit: unitChoices.TiB.toString(),
+      });
+    } else {
+      const updatedFields = fields.map((field) => ({
+        name: field.name,
+        tag: platform,
+        capacity: field.capacity,
+        capacityUnit: field.capacityUnit,
+      }));
+      replace(updatedFields);
+    }
+  }, [platform, fields.length]);
+
+  const handleBucketNumberChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const newNumber = e.target.valueAsNumber;
+
+    if (newNumber < 1 || newNumber > 20 || isNaN(newNumber)) {
+      return;
+    }
+
+    setBucketNumber(newNumber);
+
+    if (newNumber < fields.length) {
+      for (let i = fields.length - 1; i >= newNumber; i--) {
+        remove(i);
+      }
+    } else if (newNumber > fields.length) {
+      const newFields = Array(newNumber - fields.length).fill({
+        name: '',
+        tag: platform,
+        capacity: '0',
+        capacityUnit: unitChoices.TiB.toString(),
+      });
+      append(newFields);
+    }
+  };
+
+  const bucketNamePlaceholder = `${platform}-bucket-name`;
+
+  const xCoreLibrary = useXCoreLibrary();
+  const { useClusterCapacity } =
+    xCoreLibrary === XCORE_NOT_AVAILABLE ? () => ({}) : xCoreLibrary;
```

Comment:
When XCORE is unavailable I think that this will raise an error because you cannot destructurate `useClusterCapacity` from `() => ({})`